### ISSUE TITLE
Regression errors for a Clang-15 build on Linux

### ISSUE THREAD

**Issue by fbrausse** [esbmc/esbmc#1371]
When compiling a RelWithDebInfo build of ESBMC with Clang-15 on Ubuntu-22.04 using their Clang-11 package for ESBMC's C/C++ frontends, a huge amount of warnings is printed. But compilation finishes and ESBMC runs. However, a number of regression tests fail with the resulting ESBMC executable: `6 tests failed out of 3761`

```
The following tests FAILED:
	1921 - regression/bitwuzla/memset-good (Failed)
	1989 - regression/esbmc-cpp11/constructors/Constructor1-5 (Failed)
	1990 - regression/esbmc-cpp11/constructors/Constructor1-6 (Failed)
	2444 - regression/esbmc-cpp/polymorphism_bringup/llbmc_multiple_inheritance (Failed)
	2445 - regression/esbmc-cpp/polymorphism_bringup/llbmc_multiple_inheritance-wrong (Failed)
	2446 - regression/esbmc-cpp/polymorphism_bringup/llbmc_virtual_inheritance (Failed)
```
The cpp ones of those all share
```
Violated property:
  
  function call: not enough arguments


VERIFICATION FAILED
```

I've used #1370 to enable Clang compilation.

**Comment by fbrausse** [comment]
`regression/bitwuzla/memset-good` is most likely #1171. The rest are C++ ones, so I'll add the corresponding label.

**Comment by fbrausse** [comment]
Fixed by #1384 as well.
